### PR TITLE
Unit testing for mqttcm

### PR DESCRIPTION
**Pull request by suvi-123**

### CODE REVIEW

File: tests/test_mqttcm_connect.c [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+    // Check if the function returns a zero value indicating failure
+    CU_ASSERT_EQUAL(result, 0);
+
+    // Check that the IP address is not retrieved (should be NULL)
+    CU_ASSERT_PTR_NULL(ip);
+
+
+}
+
+
```
Comment (guruchandru) [inline tests/test_mqttcm_connect.c:397]: Normalize the space between different test case , some where it is 5 new lines between two successive lines and other place it is 2 or 1 newline

File: CMakeLists.txt [hunk 2]
```diff
@@ -95,6 +95,4 @@ link_directories ( ${LIBRARY_DIR} ${COMMON_LIBRARY_DIR} ${LIBRARY_DIR64} )
 
 add_subdirectory(src)
 
-#if (BUILD_TESTING)
```
Comment (sadhyama) [inline CMakeLists.txt:98]: BUILD_TESTING is required

File: src/mqttcm_connect.c [hunk 3]
```diff
@@ -56,6 +56,12 @@ int get_global_shutdown()
 	return 0;
 }
 
+int Get_mqinit()
+{
+    MqttCMInfo("Inside Get_mqinit weak function.\n");
+    mqinit = 0;
```
Comment (sadhyama) [inline src/mqttcm_connect.c:62]: Adding mqinit = 0 is not correct in getter func as the value will get changed to 1 or 0 during process flow

File: src/mqttcm_connect.c [hunk 4]
```diff
@@ -368,23 +374,26 @@ bool mqttCMConnectBroker()
...
 		pthread_cond_wait(get_global_mqtt_cond(), get_global_mqtt_mut());
 		pthread_mutex_unlock(get_global_mqtt_mut());
 		MqttCMInfo("Received mqtt signal proceed to mqtt init\n");
+		return 1;
 	}
 }
 
 int validateForMqttInit()
 {
+	printf("mqinit = %d\n",mqinit);
```
Comment (sadhyama) [inline src/mqttcm_connect.c:396]: Remove printf

File: src/mqttcm_generic.c [hunk 5]
```diff
@@ -43,24 +43,27 @@ int __attribute__((weak)) rbus_StoreValueIntoDB(char *paramName, char *value);
...
 {
     MqttCMInfo("Inside Get_Mqtt_ClientId weak function.\n");
-    return NULL;
+    return "123456789012";
 }
 int Get_Mqtt_Broker( char *pString)
 {
     MqttCMInfo("Inside Get_Mqtt_Broker weak function.\n");
-    UNUSED(pString);
+    //UNUSED(pString);
```
Comment (sadhyama) [inline src/mqttcm_generic.c:58]: Remove commented code

File: src/mqttcm_generic.c [hunk 6]
```diff
@@ -43,24 +43,27 @@ int __attribute__((weak)) rbus_StoreValueIntoDB(char *paramName, char *value);
...
+    strcpy(pString, "localhost");
     return 0;
 }
 int Get_Mqtt_Port( char *pString)
 {
     MqttCMInfo("Inside Get_Mqtt_Port weak function.\n");
-    UNUSED(pString);
+    //UNUSED(pString);
```
Comment (sadhyama) [inline src/mqttcm_generic.c:65]: Remove commented code

File: tests/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,37 @@
+set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -W -g -fprofile-arcs -ftest-coverage -O0")
```
Comment (sadhyama) [inline tests/CMakeLists.txt:1]: Add LICENCE details similar to webcfg repo

File: tests/test_mqttcm_connect.c [hunk 8]
```diff
@@ -0,0 +1,589 @@
+#include <stdio.h>
```
Comment (sadhyama) [inline tests/test_mqttcm_connect.c:1]: Add LICENCE details

File: tests/test_mqttcm_connect.c [hunk 9]
```diff
@@ -0,0 +1,589 @@
...
+/*----------------------------------------------------------------------------*/
+/*                             Test case for get_from_file                    */
+/*----------------------------------------------------------------------------*/
+
+
+
+
+
+
+void createFile(const char* filename) {
```
Comment (sadhyama) [inline tests/test_mqttcm_connect.c:124]: Better to rename this to writeFile() as this is creating and writting the file content

File: tests/test_mqttcm_connect.c [hunk 10]
```diff
@@ -0,0 +1,589 @@
...
+     mqttCMRbus_Uninit();
+
+}
+void test_checkMqttParamSet_success()
+{
+
+ 	Get_mqinit();
+    mqttCMRbusInit("componentName");
+    regMqttDataModel();
+    //Get_mqinit();
```
Comment (sadhyama) [inline tests/test_mqttcm_connect.c:221]: Remove commented code

File: tests/test_mqttcm_connect.c [hunk 11]
```diff
@@ -0,0 +1,589 @@
...
+
+
+/*----------------------------------------------------------------------------*/
+/*                             Test case for getHostIPFromInterface           */
+/*----------------------------------------------------------------------------*/
+
+
+
+void test_getHostIPFromInterface_success()
+{
```
Comment (sadhyama) [inline tests/test_mqttcm_connect.c:323]: Is this func completed? or still in progress as some of the code is commented?

File: tests/test_mqttcm_connect.c [hunk 12]
```diff
@@ -0,0 +1,589 @@
...
+/*                             Test case for get_interface                    */
+/*----------------------------------------------------------------------------*/
+
+
+
+void test_get_interface_success() {
+    const char* devicePropsFile = "/tmp/device.properties";
+    const char* content = "WEBCONFIG_INTERFACE=eth0\n";
+
+    FILE* file = fopen(devicePropsFile, "w");
```
Comment (sadhyama) [inline tests/test_mqttcm_connect.c:410]: can we reuse same createFile functions here?

File: tests/test_mqttcm_connect.c [hunk 13]
```diff
@@ -0,0 +1,589 @@
...
+
+
+
+void test_executeMqttScript_success()
+{
+
+ char scriptName[] = "./script.sh";
+   FILE* scriptFile = fopen(scriptName, "w+");
+   if (scriptFile == NULL) {
+    //CU_FAIL("Failed to create script file");
```
Comment (sadhyama) [inline tests/test_mqttcm_connect.c:465]: Remove it

File: tests/CMakeLists.txt (hunk 7 above)
Comment (sadhyama) [reply tests/CMakeLists.txt:15]: Use same as mqttcm_connect.c , test_mqttcm_connect.c

File: tests/test_mqttcm_connect.c [hunk 14]
```diff
@@ -0,0 +1,589 @@
...
+    // Wait for the signaling thread to finish
+    pthread_join(thread, NULL);
+    CU_ASSERT_EQUAL(result, 1);
+     mqttCMRbus_Uninit();
+
+}
+void test_checkMqttParamSet_success()
+{
+
+ 	Get_mqinit();
```
[the remark this answers is not in the document]
Comment (sadhyama) [reply tests/test_mqttcm_connect.c]: Resolving as Get_mqinit is removed now

**Review by sadhyama — APPROVED** [review]

**Pull request merged**